contracts/contracts/strategies/ConvexOUSDMetaStrategy.sol

```diff
@@ -0,0 +1,222 @@
+import { IERC20 } from "./BaseCurveStrategy.sol";
+import { BaseConvexMetaStrategy } from "./BaseConvexMetaStrategy.sol";
+import { StableMath } from "../utils/StableMath.sol";
+import { IVault } from "../interfaces/IVault.sol";
+
+contract ConvexOUSDMetaStrategy is BaseConvexMetaStrategy {
+    using StableMath for uint256;
+    using SafeERC20 for IERC20;
+
+    /* Take 3pool LP and mint the corresponding amount of ousd. Deposit and stake that to
+     * ousd Curve Metapool. Take the LP from metapool and deposit them to Convex.
+     */
+    function _lpDepositAll() internal override {
+        IERC20 threePoolLp = IERC20(pTokenAddress);
+        IERC20 metapoolErc20 = IERC20(address(metapool));
+        ICurvePool curvePool = ICurvePool(platformAddress);
+
+        uint256 threePoolLpBalance = threePoolLp.balanceOf(address(this));
+        uint256 curve3PoolVirtualPrice = curvePool.get_virtual_price();
+        uint256 threePoolLpDollarValue = threePoolLpBalance.mulTruncate(
+            curve3PoolVirtualPrice
+        );
+
+        uint128 crvCoinIndex = _getMetapoolCoinIndex(pTokenAddress);
+        uint128 ousdCoinIndex = _getMetapoolCoinIndex(
+            address(metapoolMainToken)
+        );
+
+        uint256 ousdToAdd = toPositive(
+            int256(
+                metapool.balances(crvCoinIndex).mulTruncate(
+                    curve3PoolVirtualPrice
+                )
+            ) -
+                int256(metapool.balances(ousdCoinIndex)) +
+                int256(threePoolLpDollarValue)
+        );
+
+        /* Mint so much ousd that the dollar value of 3CRVLP in the pool and OUSD will be equal after
+         * deployment of liquidity. In cases where pool is heavier in OUSD before the deposit strategy mints
+         * less OUSD and gets less metapoolLP and less rewards. And when pool is heavier in 3CRV strategy mints
+         * more OUSD and gets more metapoolLP and more rewards.
+         *
+         * In both cases metapool ends up being balanced and there should be no incentive to execute arbitrage trade.
+         */
+        if (ousdToAdd > 0) {
+            IVault(vaultAddress).mintForStrategy(ousdToAdd);
+        }
+
+        uint256 ousdBalance = metapoolMainToken.balanceOf(address(this));
+        uint256[2] memory _amounts = [ousdBalance, threePoolLpBalance];
+
+        uint256 metapoolVirtualPrice = metapool.get_virtual_price();
+        /**
+         * First convert all the deposited tokens to dollar values,
+         * then divide by virtual price to convert to metapool LP tokens
+         * and apply the max slippage
+         */
+        uint256 minReceived = (ousdBalance + threePoolLpDollarValue)
+            .divPrecisely(metapoolVirtualPrice)
+            .mulTruncate(uint256(1e18) - maxSlippage);
+
+        // slither-disable-next-line unused-return
+        metapool.add_liquidity(_amounts, minReceived);
+
+        uint256 metapoolLp = metapoolErc20.balanceOf(address(this));
+
```

Comment:
Super tiny: The balanceOf and the deposit line are tightly linked since the amount is used in the deposit and never used elsewhere. This means that we probably don't need a new line separating these two lines.

Reply:
👍